cpp/core/compute/ExecutionCtx.h

```diff
@@ -178,7 +202,7 @@ struct ExecutionCtxFactoryContext {
     backendFactory = factory;
   }
 
-  std::shared_ptr<ExecutionCtx> create() {
+  ExecutionCtx* create() {
```

Comment:
Would it be better to make `ExecutionCtxFactoryContext` a class as we have mutex to be hidden from callers?

Reply:
Could you explain more about that?

Reply:
I meant to use `class` rather than `struct` on `ExecutionCtxFactoryContext`.

Reply:
Worth to try, let's try that in the following patch.